packages/eas-cli/src/utils/fingerprintCli.ts

```diff
@@ -0,0 +1,36 @@
+import { Env, Workflow } from '@expo/eas-build-job';
+import { silent as silentResolveFrom } from 'resolve-from';
+
+export async function createFingerprintAsync(
+  projectDir: string,
+  options: {
+    workflow: Workflow;
+    platform: string;
+    debug?: boolean;
+    env: Env | undefined;
+    cwd?: string;
+  }
+): Promise<{
+  hash: string;
+  sources: object[];
+  isDebugSource: boolean;
+} | null> {
+  // @expo/fingerprint is exported in the expo package for SDK 52+
+  const fingerprintPath = silentResolveFrom(projectDir, 'expo/fingerprint');
+  if (!fingerprintPath) {
+    return null;
+  }
+  const Fingerprint = require(fingerprintPath);
+  const fingerprintOptions: Record<string, any> = {};
+  if (options.platform) {
+    fingerprintOptions.platforms = [options.platform];
+  }
+  if (options.workflow === 'managed') {
+    fingerprintOptions.ignorePaths = ['android/**/*', 'ios/**/*'];
+  }
```

Comment:
Not for this PR, but I wonder if it makes sense to hoist the workflow discrimination into the fingerprint library itself? (So that this stays consistent with https://github.com/expo/expo/blob/main/packages/expo-updates/utils/src/createFingerprintAsync.ts#L22). I don't foresee the logic changing, but it does seem fairly core to what fingerprint selects. Maybe make it an argument/option that, when provided, sets these ignore paths by default? cc @Kudo

Reply:
sounds good to me. maybe we can add an `isCNG: boolean` option.